txwrapplog.py

```diff
@@ -22,8 +23,10 @@
 }
 
 
-def wrapp_observer(output):
+def wrapp_observer(output, service=None, host=None):
     json = jsonFileLogObserver(output, recordSeparator='')
+    service = service or os.environ.get('SERVICE_NAME')
+    host = host or os.environ.get('HOSTNAME')
 
     @provider(ILogObserver)
     def wrapped(event):
```

Comment:
I'm not sure. I did not really work on that part of the code. It was already there.

More information here:
https://docs.zope.org/zope.interface/api.html#zope.interface.declarations.provider